src/metabase/pulse/render/body.clj

```diff
@@ -1030,16 +1030,31 @@
       [:img {:style (style/style {:display :block :width :100%})
              :src   (:image-src image-bundle)}]]}))
 
+(defn- all-unique?
+  [funnel-rows]
+  (let [ks (into #{} (map :key funnel-rows))]
+    (= (count ks) (count funnel-rows))))
+
+(defn- build-funnel-rows
```

Comment:
A minor note here, but in Clojure the approach to putting verbs in function names is slightly different than in a lot of other languages.

Quoting from the [Elements of Clojure](https://leanpub.com/elementsofclojure/read_sample) on Naming Functions

> If a function only transforms data, we should avoid verbs wherever possible. A function that calculates an MD5 hash, defined in our payload namespace, should be called md5.

Essentially the recommendation is to reserve verbs for functions that do things like fetching, mutating, publishing, etc, and use "declarative" names otherwise.

Just sharing this because I'm trying to break the habit of sticking verbs on every function myself. In this case the verb is better than shadowing the binding of the same name though.